### PR TITLE
Updated readme

### PR DESCRIPTION
**Pull request by naomilwx**
Rewrote readme to provide more context and explanation of the sample structures

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,74 @@
...
+
+### autograde.py
+Each test to be executed on the students' code submissions must be written as methods of a class extending `unittest.TestCase` as required by the unit test framework.
+
+The class must have the `setUp` instance method defined in the following manner:
+```
+def setUp(self):
+        self.meta = { 'expression': '', 'expected': '', 'hint': '' }
+```
+Public test cases are instance methods with names of the format: test_public_*
```
Comment (fonglh) [inline README.md]: Mark the code segments with back ticks `test_public_*`

File: README.md [hunk 2]
```diff
@@ -1,15 +1,74 @@
...
+
+Private test cases are instance methods with names of the format: test_private_*
+
```
Comment (fonglh) [inline README.md]: Add description for evaluation tests. `test_evaluation_*`

**Review by fonglh — APPROVED** [review]
Just some minor edits near the end.

File: README.md (hunk 2 above)
Comment (naomilwx) [reply README.md]: what are evaluation tests?

Comment (fonglh) [reply README.md]: They're used to figure out a preliminary grade for the question. So if the student gets half of the evaluation tests wrong, the system suggests that he gets half the max score of the question.

They're also totally invisible to the students, so they could pass public and private tests, but still get some edge cases under evaluation tests wrong.

Comment (naomilwx) [reply README.md]: ok updated
